app/models/setting.rb

```diff
@@ -198,10 +208,12 @@ def self.convert_array_to_regexp(array)
 
   def self.create_existing(s, opts)
     bypass_readonly(s) do
+      encrypted = opts.delete(:encrypted) || false
       attrs = column_check([:default, :description, :full_name])
       to_update = Hash[opts.select { |k,v| attrs.include? k }]
       to_update.merge!(:value => SETTINGS[opts[:name].to_sym]) if SETTINGS.key?(opts[:name].to_sym)
       s.update_attributes(to_update)
+      s.update_column :encrypted, encrypted if !column_check([:encrypted]).empty?
```

Comment:
This isn't using an ActionController::Parameters instance (i.e. user input from a controller) so mass assignment should work fine.